oasis/main.setup.ml.in

```diff
@@ -0,0 +1,13 @@
+
+let () =
+  add_variable "build_id"
+    ~doc:"Adds a build id to bap version"
+    ~define:(function
+        | Some x -> x
+        | None ->
+          try
+            ignore @@ OASISFileUtil.which ~ctxt "git";
+            OASISExec.run_read_one_line ~ctxt
+              ~f_exit_code:ignore
+              "git" ["rev-parse"; "--short=7"; "HEAD"; "2>/dev/null" ]
```

Comment:
this redirection is not portable, use the following command, which will return an empty string in case of an error and the required shorted-sha if HEAD exists:
```
git rev-parse --verify --quiet --short=7 "HEAD"
```